lib/common/event-emitter.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * In this file we effectively clone the EventEmitter class but in a way
+ * such that prototype modifications on the original EventEmitter do
+ * not propogate to Electron's internal instances.  This means that user
+ * code can't pollute Electron logic.
+ */
+import { EventEmitter as NodeEventEmitter } from 'events'
+
+export const EventEmitter = function EventEmitter (this: NodeEventEmitter) {
+  NodeEventEmitter.call(this)
+} as any as typeof NodeEventEmitter
+
+Object.assign(EventEmitter.prototype, NodeEventEmitter.prototype)
```

Comment:
Isn't this still easily modifiable from user code by:

```js
require('electron').BrowserWindow.prototype.emit = ...
```

Reply:
It's been a while but I think I was trying to avoid userland modules that mutate `EventEmitter` breaking internal electron logic.

Some module I was using was extending `EventEmitter` in a way that broke ipc.

Reply:
Can you add comment for the line in the file?

Reply:
How does Node handle this with their primordials..?

Reply:
@nornagon  Node does nothing to prevent users to change internal states, they just use `EventEmitter` and say do not modify things:
https://github.com/nodejs/node/blob/master/lib/net.js#L35
https://nodejs.org/api/http.html#http_agent_requests

Should we just keep things as they are since if users mess up with `EventEmitter` they would break Node anyway?

Reply:
@zcbenz if Node would also be broken by this, then it seems reasonable to me that we should be broken by it too. I'm happy to leave things as-is then.